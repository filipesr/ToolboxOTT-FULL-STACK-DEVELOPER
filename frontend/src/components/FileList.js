import { Table } from "react-bootstrap";

function FileList({ setFile, listFiles }) {
  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>File Name</th>
        </tr>
      </thead>
      <tbody style={{ cursor: "pointer" }}>
        {listFiles.map((file, key) => (
          <tr key={key} onClick={() => setFile(file)}>
            <td>{file}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
}

export default FileList;
