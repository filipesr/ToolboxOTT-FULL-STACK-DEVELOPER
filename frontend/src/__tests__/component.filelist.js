import React from "react";
import FileList from "../components/FileList";
import { create } from "react-test-renderer";

describe("FileList coponent", () => {
  const listFiles = ["teste"];
  const setFile = () => null;
  test("testing list files", () => {
    let tree = create(<FileList listFiles={listFiles} setFile={setFile} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
