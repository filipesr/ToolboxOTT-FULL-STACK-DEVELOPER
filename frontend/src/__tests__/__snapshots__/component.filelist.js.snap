// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList coponent testing list files 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        File Name
      </th>
    </tr>
  </thead>
  <tbody
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <tr
      onClick={[Function]}
    >
      <td>
        teste
      </td>
    </tr>
  </tbody>
</table>
`;
